src/all/tachidesk/src/eu/kanade/tachiyomi/extension/all/tachidesk/Tachidesk.kt

```diff
@@ -123,15 +92,27 @@ class Tachidesk : ConfigurableSource, UnmeteredSource, HttpSource() {
             }
         }
 
-        private fun Response.isUnauthorized(): Boolean = this.code == 401
+        private fun Response.isUnauthorized(): Boolean = this.code == 401 || this.isGraphQLUnauthorized()
+        private fun Response.isGraphQLUnauthorized(): Boolean {
+            @Serializable
+            data class Error(val message: String)
+
+            @Serializable
+            data class Outer(val errors: List<Error>)
+
+            return try {
+                val body = json.decodeFromStream<Outer>(this.peekBody(Long.MAX_VALUE).byteStream())
```

Comment:
Do we need to `.use` this stream to make sure it doesn't cause a memory leak?

Reply:
Just make sure to do a quick test. I don't think it should close the underlying body, but it's good to verify.